From 424bc4bcf75cebc86b693b4b920ee686ce6550ea Mon Sep 17 00:00:00 2001
From: valid-ptr <konstantin.matveyev@eligovision.ru>
Date: Wed, 17 May 2023 20:07:22 +0300
Subject: [PATCH 370/458] SDL emscripten port: preventDefault should not be
 called on mousedown. Otherwise mouseup will not be fired outside iframe in
 Chrome-based browsers

---
 src/video/emscripten/SDL_emscriptenevents.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/video/emscripten/SDL_emscriptenevents.c b/src/video/emscripten/SDL_emscriptenevents.c
index 82ce997f9..e08d06b80 100644
--- a/src/video/emscripten/SDL_emscriptenevents.c
+++ b/src/video/emscripten/SDL_emscriptenevents.c
@@ -649,6 +649,7 @@ static EM_BOOL Emscripten_HandleMouseButton(int eventType, const EmscriptenMouse
     Uint8 sdl_button_state;
     SDL_EventType sdl_event_type;
     double css_w, css_h;
+    SDL_bool prevent_default = SDL_FALSE; /* needed for iframe implementation in Chrome-based browsers. */
 
     switch (mouseEvent->button) {
     case 0:
@@ -673,6 +674,7 @@ static EM_BOOL Emscripten_HandleMouseButton(int eventType, const EmscriptenMouse
     } else {
         sdl_button_state = SDL_RELEASED;
         sdl_event_type = SDL_MOUSEBUTTONUP;
+        prevent_default = SDL_GetEventState(sdl_event_type) == SDL_ENABLE;
     }
     SDL_SendMouseButton(window_data->window, 0, sdl_button_state, sdl_button);
 
@@ -683,7 +685,7 @@ static EM_BOOL Emscripten_HandleMouseButton(int eventType, const EmscriptenMouse
         return 0;
     }
 
-    return SDL_GetEventState(sdl_event_type) == SDL_ENABLE;
+    return prevent_default;
 }
 
 static EM_BOOL Emscripten_HandleMouseFocus(int eventType, const EmscriptenMouseEvent *mouseEvent, void *userData)
-- 
2.17.1

